pilot/pkg/proxy/envoy/v2/mesh.go

```diff
@@ -0,0 +1,990 @@
+	}
+	hasUID := false
+	svcFQDNs := []string{}
+	svcDomains := []string{}
+	miscLabels := make(map[string]string, len(labels))
+	for _, label := range labels {
+		if prohibitedAttrSet[label.Name] {
+			errs = multierror.Append(fmt.Errorf(
+				"prohibited use of Istio destination label '%s' for endpoint label", label.Name))
+			continue
+		}
+		switch label.Name {
+		case DestinationService.AttrName():
+			svcFQDNs = append(svcFQDNs, label.Value)
+		case DestinationDomain.AttrName():
+			svcDomains = append(svcDomains, label.Value)
+		case DestinationUID.AttrName():
+			hasUID = true
+			fallthrough
+		default:
+			oldValue, found := miscLabels[label.Name]
+			if found {
+				errs = multierror.Append(fmt.Errorf(
+					"single value label '%s' has multiple values ['%s','%s']", label.Name, oldValue, label.Value))
+				continue
+			}
+			miscLabels[label.Name] = label.Value
+		}
+	}
+	if !hasUID {
+		errs = multierror.Append(fmt.Errorf(
+			"missing Endpoint mandatory label '%s'", DestinationUID.AttrName()))
+	}
+	if len(svcFQDNs) == 0 {
+		errs = multierror.Append(fmt.Errorf(
+			"missing Endpoint mandatory label '%s'", DestinationService.AttrName()))
+	}
+	if errs != nil {
+		return nil, multierror.Prefix(errs, "Mesh endpoint creation errors")
+	}
+	var epSocketProtocol xdsapi.SocketAddress_Protocol
+	switch socketProtocol {
+	case SocketProtocolTCP:
+		epSocketProtocol = xdsapi.SocketAddress_TCP
+	case SocketProtocolUDP:
+		epSocketProtocol = xdsapi.SocketAddress_UDP
+	}
+	miscLabels[DestinationIP.AttrName()] = ipAddr.String()
+	miscLabels[DestinationPort.AttrName()] = strconv.Itoa((int)(port))
+	ep := Endpoint{
+		Endpoint: &xdsapi.Endpoint{
+			Address: &xdsapi.Address{
+				Address: &xdsapi.Address_SocketAddress{
+					&xdsapi.SocketAddress{
+						Address:    address,
+						Ipv4Compat: ipAddr.To4() != nil,
+						Protocol:   epSocketProtocol,
+						PortSpecifier: &xdsapi.SocketAddress_PortValue{
+							PortValue: port,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Populate endpoint labels.
+	ep.setSingleValuedAttrs(miscLabels)
+	// Sort for stable comparisons down the line.
+	sort.Strings(svcFQDNs)
+	ep.setMultiValuedAttrs(DestinationService.AttrName(), svcFQDNs)
+	sort.Strings(svcDomains)
+	ep.setMultiValuedAttrs(DestinationDomain.AttrName(), svcDomains)
+	return &ep, nil
+}
+
+// determineRuleType returns the type of destination rule, i.e. how to interpret
+// the ruleName. It returns the value to use for querying subsets, ex: for a
+// ruleName with a DNS wildcard, it returns the suffix to use for the domain.
+// For an IP address, it returns a normalized address. For CIDRs, the query
+// value is set to nil, but a IP network corresponding to the CIDR specified in
+// ruleName is returned. For all other types, the returned IP Network is nil.
+func determineRuleType(ruleName string) (DestinationRuleType, string, *net.IPNet) {
+	if strings.HasPrefix(ruleName, wildCardDomainPrefix) {
+		return DestinationRuleService, ruleName[2:], nil
```

Comment:
`ruleName[len(wildCardDomainPrefix):]`

Reply:
now using the right regex. I may not have introduced constants for * and *- though.